src/libexpr-c/nix_api_value.cc

```diff
@@ -306,7 +306,7 @@ int64_t nix_get_int(nix_c_context * context, const nix_value * value)
     try {
         auto & v = check_value_in(value);
         assert(v.type() == nix::nInt);
-        return v.integer();
+        return v.integer().value;
```

Comment:
Can't `v.integer()` return `NixInt::Inner`? It would make the diff a lot smaller.

Alternatively, there could be an implicit coercion from `NixInt` to `NixInt::Inner`.

Reply:
This is deliberately explicit so that it's impossible to deal with values originating from user code without doing safe arithmetic.

Without this deliberate break, I would not have caught the broken JSON conversions from uint64_t and others.

Reply:
I like that, but it isn't quite obvious at the usage sites.
Should it be `.integer().toUnchecked()`?